src/tokio/fs_storage.rs

```diff
@@ -21,11 +25,34 @@ impl FsStorage {
     /// # Panics
     ///
     /// If there is not a tokio runtime available
-    pub fn open(root: std::path::PathBuf) -> Result<Self, std::io::Error> {
+    pub fn open<P: AsRef<Path>>(root: P) -> Result<Self, std::io::Error> {
         let handle = tokio::runtime::Handle::current();
         let inner = Arc::new(Mutex::new(crate::fs_store::FsStore::open(root)?));
         Ok(Self { inner, handle })
     }
+
+    /// Overrides the tmpdir directory used for temporary files.
+    ///
+    /// The default is to use the root directory passed to [`FsStorage::open`].
+    ///
+    /// The tmpdir used must be on the same mount point as the root directory,
+    /// otherwise the storage will throw an error on writing data.
+    ///
+    /// # Errors
+    ///
+    /// This will attempt to create the tmpdir directory and throw an error if
+    /// it does not exist.
+    pub fn with_tmpdir<P: AsRef<Path>>(self, tmpdir: P) -> Option<Result<Self, std::io::Error>> {
```

Comment:
Is this return type intuitive? `None` is only returned when we can't take ownership of the inner `FsStore`...